src/main/java/com/powsybl/openloadflow/equations/InjectionDerivable.java

```diff
@@ -0,0 +1,48 @@
+package com.powsybl.openloadflow.equations;
+
+import com.powsybl.math.matrix.DenseMatrix;
+import com.powsybl.openloadflow.network.ElementType;
+import com.powsybl.openloadflow.util.Derivable;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public class InjectionDerivable<V extends Enum<V> & Quantity> implements Derivable<V> {
+
+    private final Equation<V, ?> equation;
+
+    public InjectionDerivable(Equation<V, ?> equation) {
+        Objects.requireNonNull(equation);
+        this.equation = equation;
+    }
+
+    private Stream<? extends EquationTerm<V, ?>> getBranchTermStream() {
+        return equation.getTerms().stream().filter(EquationTerm::isActive)
+                .filter(t -> t.getElementType() == ElementType.BRANCH);
+    }
+
+    @Override
+    public double der(Variable<V> variable) {
+        return -getBranchTermStream().mapToDouble(t -> t.der(variable)).sum();
```

Comment:
Maybe you should explain why you take minus here, it might not be evident for a new comer to the project.

Reply:
@geofjamg after explaining the minus sign, I decided to change the eval implementations and add the equation RHS.
In this way the  class  is expected to return the real injection *value*.
Let me know if you agree.

Reply:
Reverted and commented. That was a bad idea.